openfasoc/generators/glayout/glayout/flow/blocks/elementary/cascode_common_source/cascode_common_source.py

```diff
@@ -0,0 +1,407 @@
+
+from gdsfactory.components import text_freetype, rectangle
+from gdsfactory import Component
+from gdsfactory.routing.route_quad import route_quad
+from glayout.flow.spice.netlist import Netlist
+from typing import Optional, Union 
+
+
+from glayout.flow.placement.two_transistor_interdigitized import two_nfet_interdigitized, two_pfet_interdigitized
+from glayout.flow.pdk.mappedpdk import MappedPDK
+from glayout.flow.routing.c_route import c_route
+from glayout.flow.routing.L_route import L_route
+from glayout.flow.routing.straight_route import straight_route
+from glayout.flow.routing.smart_route import smart_route
+from glayout.flow.spice.netlist import Netlist
+from glayout.flow.pdk.sky130_mapped import sky130_mapped_pdk as sky130
+from glayout.flow.primitives.fet import nmos, pmos
+from glayout.flow.primitives.guardring import tapring
+from glayout.flow.pdk.util.port_utils import add_ports_perimeter	
+from gdsfactory.component import Component
+from typing import Optional, Union 
+from gdsfactory.components import text_freetype, rectangle
+
+from glayout.flow.pdk.util.comp_utils import prec_ref_center, prec_center, movey, evaluate_bbox, align_comp_to_port
+
+from glayout.flow.primitives.via_gen import via_array
+
+# Netlist for LVS and SPICE simulations
+def cascode_common_source_netlist(
+	pdk: MappedPDK, 
+	m1_width: float,
+	m2_width: float,
+	m1_length: float,
+	m2_length: float,
+	multipliers: int, 
+	n_or_p_fet: Optional[str] = 'nfet',
+	subckt_only: Optional[bool] = False,
+	m1_fingers = int,
+	m2_fingers = int,
+	m1_multipliers = int,
+	m2_multipliers = int
+) -> Netlist:
+	if m1_length is None:
+		m1_length = pdk.get_grule('poly')['min_length']
+	if m1_width is None:
+		m1_width = pdk.get_grule('poly')['min_width']
+	m2_length = m2_length or pdk.get_grule('poly')['min_length']
+	m2_width = m2_width or pdk.get_grule('poly')['min_width']
+
+	mtop = multipliers if subckt_only else 1
+	model = pdk.models[n_or_p_fet]
+	m1_multipliers = m1_multipliers or 1
+	m2_multipliers = m2_multipliers or 1
+	dmtop = m1_fingers*m1_multipliers
+	num_dummies = 4
+	
+	circuit_name='CASCODECOMMONSRC'
+	nodes=['VIN', 'VBIAS', 'VSS', 'IOUT', "INT"]
+	model= model
+	m1_width= m1_width
+	m2_width= m2_width
+	m1_length= m1_length
+	m2_length= m2_length
+	mult= m1_multipliers*m1_fingers
+	m1_multipliers= m1_multipliers
+	m2_multipliers= m2_multipliers
+
+	source_netlist = f".subckt {circuit_name} {' '.join(nodes)}\n" 
+	# M1 and M2 transistor NETLIST
+	source_netlist += f"XM1 INT VIN VSS VSS {model} l={m1_length} w={m1_width} m={mult}\n"
+	source_netlist += f"XM2 IOUT VBIAS INT VSS {model} l={m2_length} w={m2_width} m={mult}"
+	#Adding the dummies
+	# for i in range(num_dummies):
+	# 	source_netlist += """ \nXDUMMY"""+f'{i+1}'+""" VSS VSS VSS VSS {model} """+f'l={m1_length} w={m1_width} m={1} dm={dmtop}'
+
+	source_netlist += "\n.ends {circuit_name}"
+
+	instance_format = "X{name} {nodes} {circuit_name} l={length} w={width} m={mult}"
+ 
+	return Netlist(
```

Comment:
You do not need to construct the whole netlist. You can use the sub-devices (the NMOS) and use python to construct the netlist.

https://github.com/idea-fasoc/OpenFASOC/blob/e7f461123dd9e95216ac9cd59f057c8922fe47de/openfasoc/generators/glayout/glayout/flow/spice/netlist.py#L142